### PR TITLE
Add documentation for VBS in CngKeyCreationOptions

### PR DESCRIPTION
**Pull request by vcsjones**
This is more or less a repeat of the Win32 docs, trimmed down a bit.

### CODE REVIEW

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs [hunk 1]
```diff
@@ -15,8 +15,21 @@ public enum CngKeyCreationOptions : int
...
         PreferVbs = 0x00010000,             // NCRYPT_PREFER_VBS_FLAG
+
+        /// <summary>
+        /// Indicates a key must be protected with Virtualization-based security (VBS).
+        /// </summary>
         RequireVbs = 0x00020000,            // NCRYPT_REQUIRE_VBS_FLAG
+
+        /// <summary>
+        /// An additional flag that can be used along with <see cref="RequireVbs"/> or <see cref="PreferVbs"/> .
+        /// Instructs Virtualization-based security (VBS) to protect the client key with a per-boot key.
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs:31]: I think the "useful part" should be first, so

```suggestion
        /// Instructs Virtualization-based security (VBS) to protect the client key with a per-boot key
        /// when combined with <see cref="RequireVbs"/> or <see cref="PreferVbs"/> .

```

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs [hunk 2]
```diff
@@ -15,8 +15,21 @@ public enum CngKeyCreationOptions : int
...
         MachineKey = 0x00000020,            // NCRYPT_MACHINE_KEY_FLAG
         OverwriteExistingKey = 0x00000080,  // NCRYPT_OVERWRITE_KEY_FLAG
+
+        /// <summary>
+        /// Indicates a key should be protected with Virtualization-based security (VBS).
+        /// </summary>
         PreferVbs = 0x00010000,             // NCRYPT_PREFER_VBS_FLAG
+
+        /// <summary>
+        /// Indicates a key must be protected with Virtualization-based security (VBS).
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs:25]: For all of these, should it be "Virtualization-based security (VBS)" or "Virtualization-Based Security (VBS)"? (I feel like things being extracted to initialisms usually use title case... or rather "the extracted letters" case)

Comment (vcsjones) [reply src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs:25]: I cased it the same way the win32 docs did, and the article on VBS does not title case it, except when actually used as a title.

* https://learn.microsoft.com/en-us/windows-hardware/design/device-experiences/oem-vbs
* https://learn.microsoft.com/en-us/windows/win32/api/ncrypt/nf-ncrypt-ncryptcreatepersistedkey

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs [hunk 3]
```diff
@@ -15,8 +15,22 @@ public enum CngKeyCreationOptions : int
...
+        /// </summary>
         RequireVbs = 0x00020000,            // NCRYPT_REQUIRE_VBS_FLAG
+
+        /// <summary>
+        /// Instructs Virtualization-based security (VBS) to protect the client key with a per-boot key
+        /// when combined with <see cref="RequireVbs"/> or <see cref="PreferVbs"/> .
+
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs:32]:
```suggestion
        /// when combined with <see cref="RequireVbs"/> or <see cref="PreferVbs"/> .
```

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs [hunk 4]
```diff
@@ -15,8 +15,22 @@ public enum CngKeyCreationOptions : int
...
+        /// </summary>
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs:33]:
```suggestion
        /// Instructs Virtualization-based security (VBS) to protect the client key with a per-boot key
        /// when combined with <see cref="RequireVbs"/> or <see cref="PreferVbs"/> .
        /// </summary>
```

**Review by bartonjs — APPROVED** [review]

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs [hunk 5]
```diff
@@ -15,8 +15,21 @@ public enum CngKeyCreationOptions : int
...
         PreferVbs = 0x00010000,             // NCRYPT_PREFER_VBS_FLAG
+
+        /// <summary>
+        /// Indicates a key must be protected with Virtualization-based security (VBS).
+        /// </summary>
         RequireVbs = 0x00020000,            // NCRYPT_REQUIRE_VBS_FLAG
+
+        /// <summary>
+        /// Instructs Virtualization-based security (VBS) to protect the client key with a per-boot key
+        /// when combined with <see cref="RequireVbs"/> or <see cref="PreferVbs"/> .
```
Comment (teo-tsirpanis) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKeyCreationOptions.cs:31]:
```suggestion
        /// Instructs Virtualization-based security (VBS) to protect the client key with a key that can't
        /// be reused across boot cycles when combined with <see cref="RequireVbs"/> or
        /// <see cref="PreferVbs"/>.
```

**Pull request merged**